src/sources/internal_metrics.rs

```diff
@@ -44,12 +33,15 @@ impl SourceConfig for InternalMetricsConfig {
         shutdown: ShutdownSignal,
         out: Pipeline,
     ) -> crate::Result<super::Source> {
-        Ok(Box::pin(run(
-            get_controller()?,
-            self.scrape_interval_secs,
-            out,
-            shutdown,
-        )))
+        let interval = time::Duration::from_secs(self.scrape_interval_secs);
+        if interval < time::Duration::from_millis(500) {
+            warn!(
+                "Interval set too low ({} secs), use interval >= 0.5 secs.",
```

Comment:
```suggestion
                "Interval set to 0 secs, this could result in high CPU utilization.It is suggested to use interval >= 1 secs.",
```

Given this is a `u64` now, I think using `0.5` in the warning is a bit weird. I also tried to clarify the expected bad result of using `0`.

Reply:
Thanks! Applied